### PR TITLE
Add buildspecs for RHEL 8 OVA and Raw image builds

### PR DESCRIPTION
**Pull request by abhay-krishna**
* Adding buildspecs for RHEL 8 OVA and Raw image builds.
* Adding patch to support RHEL 8 OVA builds in image-builder.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/kubernetes-sigs/image-builder/Makefile [hunk 1]
```diff
@@ -57,22 +57,32 @@ TARGETS_ALLOWED_WITH_NO_RELEASE_BRANCH=build release clean help binaries checksu
...
-BUILD_RAW_TARGETS=release-raw-ubuntu-2004-efi $(FAKE_UBUNTU_RAW_PATH) $(FINAL_UBUNTU_RAW_IMAGE_PATH) upload-artifacts-raw
+BUILD_OVA_TARGETS=setup-packer-configs-ova download-ova-bottlerocket $(FAKE_UBUNTU_OVA_PATH) $(FINAL_OVA_PATH) $(FINAL_BOTTLEROCKET_OVA_PATH) upload-artifacts-ova
+BUILD_RAW_TARGETS=release-raw-ubuntu-2004-efi $(FAKE_UBUNTU_RAW_PATH) $(FINAL_RAW_IMAGE_PATH) upload-artifacts-raw
 BUILD_TARGETS=$(BUILD_RAW_TARGETS) $(BUILD_AMI_TARGETS) $(BUILD_OVA_TARGETS)
 ifeq ($(IMAGE_FORMAT),ova)
-	S3_TARGET_PREREQUISITES=$(FINAL_UBUNTU_OVA_PATH) $(FINAL_BOTTLEROCKET_OVA_PATH)
-	RELEASE_TARGETS=release-ova-ubuntu-2004 release-ova-bottlerocket upload-artifacts-ova
+	ifeq ($(IMAGE_OS),ubuntu)
+		S3_TARGET_PREREQUISITES=$(FINAL_OVA_PATH) $(FINAL_BOTTLEROCKET_OVA_PATH)
+		RELEASE_TARGETS=release-ova-ubuntu-2004 download-ova-bottlerocket upload-artifacts-ova
```
Comment (vignesh-goutham) [inline projects/kubernetes-sigs/image-builder/Makefile:74]: Can we add a comment here why we are piggy-backing if OS is ubuntu we want to run bottlerocket download/processing. Im for sure going to think why is this here few weeks later :P 

For later, maybe we can have a layer above IMAGE_OS, something like `VENDED_OS_RELEASE_TARGETS=ubuntu bottlerocket`. Or we can separate bottlerocket and have its own buildspec.

File: projects/kubernetes-sigs/image-builder/patches/0009-Add-support-for-RHEL-8-OVA-builds.patch [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
++  "guest_os_type": "rhel8-64",
++  "iso_checksum": "48f955712454c32718dcde858dea5aca574376a1d7a4b0ed6908ac0b85597811",
++  "iso_checksum_type": "sha256",
++  "iso_url": "rhel-8.4-x86_64-dvd.iso",
++  "os_display_name": "RHEL 8",
++  "redhat_epel_rpm": "https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm",
++  "epel_rpm_gpg_key": "https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-8",
++  "shutdown_command": "sys-unconfig",
++  "vsphere_guest_os_type": "rhel7_64Guest"
++}
```
Comment (vignesh-goutham) [inline projects/kubernetes-sigs/image-builder/patches/0009-Add-support-for-RHEL-8-OVA-builds.patch:48]: dope! upstream it!

**Review by vignesh-goutham — APPROVED** [review]

File: projects/kubernetes-sigs/image-builder/Makefile (hunk 1 above)
Comment (abhay-krishna) [reply projects/kubernetes-sigs/image-builder/Makefile:74]: Will do, I kinda wanted to put bottlerocket in a separate buildspec to make it cleaner. I will do a follow-up to this.

**Review by vignesh-goutham — APPROVED** [review]
/lgtm

**Pull request merged**